### DISCUSSION TITLE
Update `print` calls to `notify` calls for `install` stage?

### DISCUSSION THREAD

**Discussion by carschandler** [nvim-treesitter/nvim-treesitter#6904] [Ideas]
I don't love when I update Treesitter and a bunch of parsers get updated (or I am reinstalling them all for some reason) and I get flooded with `print` statements that can sometimes block editing. I've been experimenting with replacing those `print` calls to `vim.notify()` or `utils.notify()` calls alongside [fidget.nvim](https://github.com/j-hui/fidget.nvim?tab=readme-ov-file#options) and it is working really nicely! The notifications just stack up in the corner without blocking.

For other notification clients that aren't quite as lightweight like `nvim-notify` or `mini-notify`, I'm not sure what this experience would be like, so maybe it could be introduced as a config option rather than as default? And if no notification client is installed, in theory the experience is virtually the same as with `print` to my understanding.

**Comment by clason** [comment]
That code is deprecated and will be removed in 1.0; the new (much improved) code uses styled output that is not supported by `vim.notify` (which itself has a doubtful future, since the whole messaging API in Neovim needs an overhaul). So for the moment (and foreseeable future) we want to stick to the very basics here.

**Reply by carschandler** [reply]
That sounds reasonable enough! Thanks for the info.
